boards/arm/cy8cproto_062_4343w/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2021 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+zephyr_include_directories(bsp/)
+zephyr_include_directories(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/)
+
+zephyr_library()
+zephyr_library_sources(board.c)
+zephyr_library_sources(bsp/cybsp.c)
+zephyr_library_sources(bsp/system_psoc6_cm4.c)
+
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_clocks.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_peripherals.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_pins.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_qspi_memslot.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_routing.c)
+zephyr_library_sources(bsp/COMPONENT_BSP_DESIGN_MODUS/GeneratedSource/cycfg_system.c)
```

Comment:
what's going on with all these autogenerated files? I guess this, at best, should be part of a HAL. Some look suspicious, though, like custom clock/pin control

Reply:
I agreed! I'll add that everything that is autogenerated should stay at vendors hal and not ay Zephyr main tree.
This is why a plan is important!

Reply:
Agreed as well, please move this to the HAL repo.